Ports/flatbuffers/patches/flatbuffers-1.12.0.patch

```diff
@@ -0,0 +1,35 @@
+diff -ruN flatbuffers-1.12.0/src/code_generators.cpp flatbuffers-1.12.0-serenity/src/code_generators.cpp
+--- flatbuffers-1.12.0/src/code_generators.cpp	2020-03-12 19:33:39.000000000 -0300
++++ flatbuffers-1.12.0-serenity/src/code_generators.cpp	2021-03-31 20:39:12.000000000 -0300
+@@ -23,6 +23,17 @@
+ #include "flatbuffers/base.h"
+ #include "flatbuffers/util.h"
+ 
++#if defined(__serenity__)
++  // We do not have those functions inside std namespace...
++
++namespace std {
++  auto isnan(double x) { return ::isnan(x); }
++  auto isinf(double x) { return ::isinf(x); }
++}
++
++#endif
++
++
+ #if defined(_MSC_VER)
+ #  pragma warning(push)
+ #  pragma warning(disable : 4127)  // C4127: conditional expression is constant
+diff -ruN flatbuffers-1.12.0/src/util.cpp flatbuffers-1.12.0-serenity/src/util.cpp
+--- flatbuffers-1.12.0/src/util.cpp	2020-03-12 19:33:39.000000000 -0300
++++ flatbuffers-1.12.0-serenity/src/util.cpp	2021-03-31 20:28:59.000000000 -0300
+@@ -189,6 +189,10 @@
+ std::string AbsolutePath(const std::string &filepath) {
+   // clang-format off
+ 
++  #ifdef __serenity__
++  #define PATH_MAX 4096
++  #endif
```

Comment:
The correct way (according to Dr. POSIX) to get `PATH_MAX` is to make sure `<limits.h>` is included.

Reply:
Oh, sorry, I forgot about `<limits.h>`...

But a weird thing: the file I patched already includes this header

```cpp
#ifdef _WIN32
#  ifndef WIN32_LEAN_AND_MEAN
#    define WIN32_LEAN_AND_MEAN
#  endif
#  ifndef NOMINMAX
#    define NOMINMAX
#  endif
#  ifdef _MSC_VER
#    include <crtdbg.h>
#  endif
#  include <windows.h> 
#  include <direct.h>
#  include <winbase.h>
#  undef interface 
#else
#  define _XOPEN_SOURCE 600  
#  include <limits.h>  // <----------
#endif
// clang-format on
```
(the complete file is [here](https://github.com/google/flatbuffers/blob/v1.12.0/src/util.cpp#L20-L38) )

------

I will try to compile a test program to investigate

Reply:
I investigated. It was something with the toolchain
I recompiled it and it worked